tests/provision/provisioning_rules.go

```diff
@@ -0,0 +1,1708 @@
+metadata:
+  name: placement-2
+spec:
+  volumeAffinity:
+  - enforcement: required
+    topologyKey: failure-domain.beta.kubernetes.io/px_zone
+    matchExpressions:
+      - key: app
+        operator: In
+        values:
+          - "mysql"
+  replicaAffinity:
+  - enforcement: required
+    topologyKey: failure-domain.beta.kubernetes.io/px_zone
+---
+apiVersion: portworx.io/v1beta2
+kind: VolumePlacementStrategy
+metadata:
+  name: placement-3
+spec:
+  volumeAffinity:
+  - enforcement: required
+    topologyKey: failure-domain.beta.kubernetes.io/px_region
+    matchExpressions:
+      - key: app
+        operator: In
+        values:
+          - "mysql"
+---
+apiVersion: portworx.io/v1beta2
+kind: VolumePlacementStrategy
+metadata:
+  name: placement-1
+spec:
+  volumeAffinity:
+  - enforcement: required
+    topologyKey: failure-domain.beta.kubernetes.io/px_zone
+    matchExpressions:
+      - key: app
+        operator: In
+        values:
+          - "mysql"
+  replicaAffinity:
+  - enforcement: required
+    topologyKey: failure-domain.beta.kubernetes.io/px_zone`
+	return vpsSpec
+}
+
+func (v *vpscase18) CleanVps() {
+	logrus.Infof("Cleanup test case context for: %v", v.name)
+}
+
+//#---- Case 19 ---- T866790
+//Verify replica affinity and volume anti-affinity topology keys with volume labels
+// keys	 with volume labels
+type vpscase19 struct {
+	//Case description
+	name string
+	// Enabled
+	enabled bool
+}
+
+func (v *vpscase19) GetLabels() ([]labelDict, int) {
+
+	lbldata := []labelDict{}
+	node1lbl := labelDict{"failure-domain.beta.kubernetes.io/px_zone": "east", "failure-domain.beta.kubernetes.io/px_region": "usa"}
+	node2lbl := labelDict{"failure-domain.beta.kubernetes.io/px_zone": "east", "failure-domain.beta.kubernetes.io/px_region": "usa"}
+	node3lbl := labelDict{"failure-domain.beta.kubernetes.io/px_zone": "east", "failure-domain.beta.kubernetes.io/px_region": "usa"}
+	node4lbl := labelDict{"failure-domain.beta.kubernetes.io/px_zone": "west", "failure-domain.beta.kubernetes.io/px_region": "usa"}
+	node5lbl := labelDict{"failure-domain.beta.kubernetes.io/px_zone": "west", "failure-domain.beta.kubernetes.io/px_region": "jp"}
+	node6lbl := labelDict{"failure-domain.beta.kubernetes.io/px_zone": "north", "failure-domain.beta.kubernetes.io/px_region": "jp"}
+	node7lbl := labelDict{"failure-domain.beta.kubernetes.io/px_zone": "north", "failure-domain.beta.kubernetes.io/px_region": "jp"}
+	node8lbl := labelDict{"failure-domain.beta.kubernetes.io/px_zone": "north", "failure-domain.beta.kubernetes.io/px_region": "jp"}
+	lbldata = append(lbldata, node1lbl, node2lbl, node3lbl, node4lbl, node5lbl, node6lbl, node7lbl, node8lbl)
+	return lbldata, 1
+}
+
+//StorageClass placement_strategy mapping
+func (v *vpscase19) GetScStrategyMap() map[string]string {
+	return map[string]string{"placement-1": "placement-1", "placement-2": "placement-2", "placement-3": ""}
+}
+
+
+func (v *vpscase19) GetSpec() string {
+
+	var vpsSpec string
```

Comment:
can we add this spec as a file?

Reply:
There are at present 24 VPS test cases and each has different spec. 
If we create all the 24 spec file, they will get activated at once instead of  during the  corresponding test case